tools/pubsys/src/kit/publish_kit/mod.rs

```diff
@@ -0,0 +1,213 @@
+    let vendor = infra_config
+        .vendor
+        .as_ref()
+        .unwrap()
+        .get(&publish_kit_args.vendor)
+        .context(error::VendorNotFoundSnafu {
+            name: publish_kit_args.vendor.clone(),
+        })?;
+    let vendor_registry_uri = vendor.registry.clone();
+    info!(
+        "Found vendor container registry at uri: {}",
+        vendor_registry_uri
+    );
+
+    // Auto resolve the expected paths for the kit contents archive
+    let kit_path = publish_kit_args.kit_path.as_path();
+    let kit_name = kit_path.file_name().unwrap().to_string_lossy();
+
+    let archives = find_oci_archives(kit_path, &kit_name)?;
+    ensure!(
+        !archives.is_empty(),
+        error::NoArchiveSnafu { path: kit_path }
+    );
+
+    let kit_version = &archives.first().unwrap().version;
+    let mut platform_images = Vec::new();
+    for archive in archives.iter() {
+        let path = kit_path.join(&archive.filename);
+
+        let out = docker!(["load", format!("--input={}", path.display()).as_str(),]);
+        let out = String::from_utf8_lossy(&out);
+        let digest_expression =
+            Regex::new("(?<digest>sha256:[0-9a-f]{64})").context(error::RegexSnafu)?;
+        let caps = digest_expression.captures(&out).unwrap();
+        let digest = &caps["digest"];
+
+        let arch_specific_target_uri = format!(
+            "{}/{}:{}-{}",
+            vendor_registry_uri, kit_name, &kit_version, &archive.arch
+        );
+
+        docker!(["tag", digest, &arch_specific_target_uri,]);
+
+        info!(
+            "Pushing kit image for platform {} to {}",
+            &archive.arch, &arch_specific_target_uri
+        );
+        docker!(["push", &arch_specific_target_uri,]);
+
+        let docker_arch = DockerArchitecture::try_from(archive.arch.as_str())
+            .context(error::InvalidArchitectureSnafu)?;
+        platform_images.push((docker_arch, arch_specific_target_uri.clone()));
+    }
+
+    let target_uri = format!("{}/{}:{}", vendor_registry_uri, kit_name, kit_version);
+
+    let images: Vec<&str> = platform_images
+        .iter()
+        .map(|(_, image)| image.as_str())
+        .collect();
+
+    let mut manifest_create_args = vec!["manifest", "create"];
+    if vendor.insecure {
+        manifest_create_args.push("--insecure")
+    }
+    manifest_create_args.push(&target_uri);
+    manifest_create_args.extend_from_slice(&images);
+    docker!(manifest_create_args);
+
+    for (arch, image) in platform_images.iter() {
+        docker!([
+            "manifest",
+            "annotate",
+            format!("--arch={}", arch).as_str(),
+            &target_uri,
+            image,
+        ]);
+    }
+
+    info!("Pushing kit to {}", &target_uri);
+    docker!(["manifest", "push", &target_uri,]);
+
+    info!("Successfully published kit to {}", target_uri);
+
+    // cleans up the transient images. these don't serve a purpose outside of publishing the kit
```

Comment:
This won't happen if we error somewhere before this. I'm trying to think of a way to make sure it happens.